app/services/record_declarations/base.rb

```diff
@@ -26,6 +26,15 @@ def profile_exists
       end
     end
 
+    def date_has_the_right_format
+      return if declaration_date.blank?
+
+      errors.add(:declaration_date, I18n.t(:invalid_declaration_date)) unless declaration_date.match(RCF3339_DATE_REGEX)
```

Comment:
so we seem to have a load translations including `invalid_declaration_date` at the root level of the locales file. these should all be namespaced accordingly. otherwise we're gonna get a very flat locale file which i think is undesired. so the line should read
```
      errors.add(:declaration_date, :invalid_declaration_date) unless declaration_date.match(RCF3339_DATE_REGEX)
```
the locale should be on something like
```
en.activemodel.errors.models.OBJECT_NAME.attributes.declaration_date.invalid_declaration_date
```
however this is probably outside the scope of this PR but something that may want addressing soon

Reply:
I will look into it in a spare moment. Ehh, I hate i18n.